src/dataset.jl

```diff
@@ -172,16 +172,133 @@ end
     write(dataset::AbstractDataset, filename::AbstractString; kwargs...)
 
 Writes the dataset to the designated filename.
+
+For the keyword arguments for vector datasets, consult the documentation of `writelayers` 
+and for raster datasets the docs for `writerasters`.
 """
 function write(
     dataset::AbstractDataset,
     filename::AbstractString;
     kwargs...,
 )::Nothing
+    
+    if nraster(dataset) > 0
+        destroy(unsafe_copy(dataset, filename = filename; kwargs...))
+    elseif nlayer(dataset) > 0
+        writelayers(dataset, filename; kwargs...)
+    elseif nraster(dataset) > 0 && nlayer(dataset) > 0
+        error("Writing datasets with raster and vector data is not supported.")
+    else
+        error("Could not determine dataset type (raster or vector).")
```

Comment:
I think it should error, because without rasters or layers it's empty, which then probably was by accident.

Reply:
GDAL does support writing empty datasets or not? If I have a vector of unknown size `n` that I want to store in a file, I'd rather not get an error for the empty case, because that may be interesting information in itself. E.g. I can create JSON files with empty arrays just fine.

Reply:
Ok, I don't mind removing it.